core/src/main/java/org/elasticsearch/index/mapper/core/TypeParsers.java

```diff
@@ -284,6 +286,15 @@ public static void parseField(FieldMapper.Builder builder, String name, Map<Stri
                 builder.fieldDataSettings(settings);
                 iterator.remove();
             } else if (propName.equals("copy_to")) {
+                if (parserContext.isWithinMultiField()) {
+                    if (indexVersionCreated.after(Version.V_2_1_0) ||
+                        (indexVersionCreated.after(Version.V_2_0_1) && indexVersionCreated.before(Version.V_2_1_0))) {
+                        throw new MapperParsingException("copy_to in multi fields is not allowed. Found the copy_to in field [" + name + "] which is within a multi field.");
+                    } else {
+                        ESLoggerFactory.getLogger("mapping [" + parserContext.type() + "]").warn("Found a copy_to in field [" + name + "] which is within a multi field. This feature has been removed and the copy_to will be ignored.");
+                        // we still parse this, otherwise the message will only appear once and the copy_to removed. After that it will appear again. Better to have it always.
+                    }
+                }
                 parseCopyFields(propNode, builder);
```

Comment:
I can do that but it might be weird with rolling upgrades. The mapping will have the copy_to until an upgraded node becomes master and then this entry suddenly vanishes and the warning stops. Seems like an undesirable behavior. 
@clintongormley and I discussed this yesterday and we thought that it is not a big deal to keep the copy_to because it will only warn whenever the mapping is changed so it will not spam the log with warnings in any case. 
If one wants to get rid of the warning they can still remove it from the mapping by updating it explicitly.